packages/prisma-client/prisma/migrations/20230106000143_tree_styles_data/migration.ts

```diff
@@ -0,0 +1,100 @@
+import { PrismaClient } from "./client";
+
+type StylesItem = {
+  breakpointId: string;
+  instanceId: string;
+  property: string;
+  value: unknown;
+};
+
+type Styles = StylesItem[];
+
+type CssRule = {
+  breakpoint: undefined | string;
+  style: Record<string, unknown>;
+};
+
+type Text = {
+  type: "text";
+  value: string;
+};
+
+type Instance = {
+  type: "instance";
+  id: string;
+  component: string;
+  cssRules: Array<CssRule>;
+  children: Array<Instance | Text>;
+};
+
+const convertTree = (instance: Instance, styles: Styles) => {
+  for (const rule of instance.cssRules) {
+    if (rule.breakpoint === undefined) {
+      continue;
+    }
+    for (const [property, value] of Object.entries(rule.style)) {
+      styles.push({
+        breakpointId: rule.breakpoint,
+        instanceId: instance.id,
+        property,
+        value,
+      });
+    }
+  }
+  instance.cssRules = [];
+  for (const child of instance.children) {
+    if (child.type === "instance") {
+      convertTree(child, styles);
+    }
+  }
+};
+
+export default async () => {
+  const client = new PrismaClient({
+    // Uncomment to see the queries in console as the migration runs
+    // log: ["query", "info", "warn", "error"],
```

Comment:
BTW, another potential reason for `JSON.stringify` error is if you've run with this line uncommented